improve rag burning & molotovs + bottles splash more than what they hit

Spookerton opened the pull request:
:cl:
tweak: Thrown bottles splash mobs around the impact atom if they break.
tweak: Rags burn with more than just fuel in them, allowing high proof alcohol, acetone, hydrazine, etc molotovs.
/:cl:

nb; molotovs will not ignite break-splashed mobs, only the hit target. If they try and pat them out, though, that'll get fun.

![https://i.imgur.com/6HvRKOc.png](https://i.imgur.com/6HvRKOc.png)

SierraKomodo commented on code/modules/reagents/reagent_containers/food/drinks/bottle.dm:
```diff
@@ -1,396 +1,446 @@
-	var/isGlass = TRUE //Whether the 'bottle' is made of glass or not so that milk cartons don't shatter when someone gets hit by it
-
-	var/obj/item/reagent_containers/glass/rag/rag = null
+	var/smash_duration = 5 /// Weaken duration for bonking heads
```
Single-line dmdocs don't parse correctly in VSC.

```suggestion
	/// Weaken duration for bonking heads
	var/smash_duration = 5
```

SierraKomodo reviewed (changes requested):
I'm not really seeing the 'splashing things other than what it hits' effect taking place. It seems to be doing the current effect of making the hit person flammable and not affecting anything else. Is there a specific condition/expected result I'm missing?

Spookerton commented on code/modules/reagents/reagent_containers/food/drinks/bottle.dm:
```diff
@@ -1,396 +1,446 @@
+					others += L
+				others -= hit_atom
+				for (var/other in others)
+					reagents.splash(other, reagents.total_volume * 0.2, copy = TRUE)
```
@SierraKomodo Here. The actual mechanic is fairly limited. Only living mobs within a 3x3 about the hit point are splashed, with a fixed non-depleting 20% of the contents of the bottle. They are *not* directly ignited if the bottle has a burning rag in it, only the hit mob is. The aim is to increase potential chaos, not direct damage (not least because fire mechanics aren't super hot). If they try to help someone who *was* ignited, they will readily catch fire because of the stacks being splashed gives them.

Spookerton committed:
rag burning uses more fuels, thrown bottles splash

strong (strength <= 15) alcohol counts as fuel
fuel, acetone, hydrazine, napalm, oil, and phoron count as fuel
rags take fuel-ness into account instead of just the fuel reagent
bottles splash mobs within 1 tile of impact when thrown and breaking
bottles *can break* when thrown on help; previously they were indestructible
bottle.isGlass -> bottle.can_shatter
cleaned up bottle file, many new -> initialize

Spookerton commented:
aaand I squashed without thinking about it. Go me.
